org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/utils/XMLPositionUtility.java

```diff
@@ -359,14 +363,119 @@ public static Range selectDTDElementDeclAt(int offset, DOMDocument document) {
 		return null;
 	}
 
-	public static Range selectDTDElementDeclTagAt(int offset, DOMDocument document) {
+	/**
+	 * Will give the range for the last VALID DTD Decl parameter at 'offset'.
+	 * An unrecognized Parameter is not considered VALID,
+	 * 
+	 * eg: <!ELEMENT elementName (content) UNRECOGNIZED_CONTENT
+	 * 		will give the range 1 character after '(content)'
+	 */
+	public static Range getLastValidDTDDeclParameter(int offset, DOMDocument document) {
 		DOMNode node = document.findNodeAt(offset);
-		if (node != null && node.isDTDElementDecl()) {
-			DTDElementDecl elementDecl = (DTDElementDecl) node;
-			return createRange(elementDecl.getStart(), elementDecl.getEndElementTag(), document);
+		if (node != null && node instanceof DTDDeclNode) {
+			DTDDeclNode decl = (DTDDeclNode) node;
+			ArrayList<DTDDeclParameter> params = decl.getParameters();
+			DTDDeclParameter finalParam;
+			if(params != null && params.size() == 0) {
+				return createRange(decl.declType.getStart(), decl.declType.getEnd(), document);
+			}
+			else if(params.get(params.size() - 1).equals(decl.unrecognized)) {

```

Comment:
always prefer constants on the left side of the equality (since it can't be null). decl.unrecognized.equals(...)